code/iaas/lb-instance/src/main/java/io/cattle/platform/lb/instance/service/impl/LoadBalancerInstanceManagerImpl.java

```diff
@@ -154,12 +173,12 @@ protected Agent getLoadBalancerAgent(LoadBalancer loadBalancer, long hostId) {
         return instances;
     }
 
-    private List<Long> populateHosts(LoadBalancer loadBalancer, Long... hostIds) {
-        List<Long> hosts = new ArrayList<Long>();
-        if (hostIds.length == 0) {
-            hosts = lbInstanceDao.getLoadBalancerHosts(loadBalancer.getId());
+    private List<LoadBalancerHostMap> populateHostMaps(LoadBalancer loadBalancer, LoadBalancerHostMap... hostMaps) {
+        List<LoadBalancerHostMap> hosts = new ArrayList<>();
```

Comment:
we should probably rename 'hosts'.  Not sure what's a good name since we already have 'hostMaps'.
Actually, looking at this again, the purpose of this function seems a little strange.  It's not quite lazy loading, but has some similarities from line 179